### ISSUE TITLE
xwallpaper doesn't redraw on screen numbering change

### ISSUE THREAD

**Issue by themad** [stoeckmann/xwallpaper#21]
I've got two monitors with different resolutions and set the background with `xwallpaper --daemon --output DP2 --maximize image1.jpg --output HDMI1 --maximize image2.jpg` and so on. 
 Sometimes my system boots with them having the wrong order, as in screen 1 is on the right. I can just change that, not problem, but as the X output size doesn't change, the xwallpaper demon doesn't redraw the pictures and it ends up looking broken. 
So it would be nice if xwallpaper either a) listens to the numbering change or b) just redraws when receiving SIGHUP or something.

**Comment by stoeckmann** [comment]
Hi themad,

Can you run xwallpaper with --debug and show me the log lines if it happens? You should see something like "event received: response_type=%u, sequence=%u" hopefully.

If you see it, it would be interesting to know how often it is printed. In case you can compile xwallpaper on your own, you might adjust the function process_event in main.c to always process the screen, regardless of size changes.

If that works for you, I have a good indication what has to be adjusted.

**Comment by stoeckmann** [comment]
Just for clarification though: You write that the displays are in different order on system boot. How is xwallpaper still running in daemon mode then? Are we talking about a suspend/resume effect here?

**Comment by themad** [comment]
Maybe I should be more verbose. This is a laptop, which explains my long xwallpaper invocation. When I switch it on while it sits in its dock, X11 autodetects my displays when it starts, but it puts them in the wrong order, i.e. the monitor connected to DP2 is on the left side, but X11 puts it to the right.
On login I run xwallpaper like this: `xwallpaper --daemon --output LVDS1 --maximize "$(shuf -en1 ~/wallpaper/*)" --output DP2 --maximize "$(shuf -en1 ~/wallpaper/*)" --output VGA1 --maximize "$(shuf -en1 ~/wallpaper/*)" --output HDMI1 --maximize "$(shuf -en1 ~/wallpaper/*)" --output DP1 --maximize "$(shuf -en1 ~/wallpaper/*)" --output HDMI2 --maximize "$(shuf -en1 ~/wallpaper/*)"`, so that I can get a random picture on each monitor. This works, but alas, the monitors are switched. So I run a script to fix the display setup. 
xwallpaper --debug output:
```
loading /home/themad/wallpaper/pic1.jpg
loading /home/themad/wallpaper/pic2.png
loading /home/themad/wallpaper/pic3.jpg
loading /home/themad/wallpaper/pic4.jpg
loading /home/themad/wallpaper/pic5.jpg
output detected: VGA1, 1680x1050+1280+0
output detected: DP2, 1920x1200+2960+0
(randr) screen dimensions: 4880x1200+0+0
creating pixmap (4880x1200)
xwallpaper: output LVDS1 was not found/disconnected, ignoring
composing /home/themad/wallpaper/pic1.jpg for DP2 (area 1920x1200+0+0) (mode 3)
xcb image (1920x1200) to DP2 (1920x1200+2960+0)
put image request parameters:
maximum request length allowed for server (32 bits): 4194303
maximum length for row data: 16777188
length of rows in image: 7680
maximum height to send: 2184
put image (1920x1200+0+0) to DP2 (1920x1200+2960+0)
composing /home/themad/wallpaper/pic2.png for VGA1 (area 1680x1050+0+0) (mode 3)
xcb image (1680x1050) to VGA1 (1680x1050+1280+0)
put image request parameters:
maximum request length allowed for server (32 bits): 4194303
maximum length for row data: 16777188
length of rows in image: 6720
maximum height to send: 2496
put image (1680x1050+0+0) to VGA1 (1680x1050+1280+0)
xwallpaper: output HDMI1 was not found/disconnected, ignoring
xwallpaper: output DP1 was not found/disconnected, ignoring
xwallpaper: output HDMI2 was not found/disconnected, ignoring
```
and here I run my "dock" script to properly set up the displays. It looks like `xrandr --output LVDS1 --off; sleep 1; xrandr --output DP2 --auto --primary --output VGA1 --right-of DP2 --auto`.
Further output of xwallpaper after that:
```
event received: response_type=89, sequence=30
output detected: VGA1, 1680x1050+0+0
output detected: DP2, 1920x1200+1680+0
(randr) screen dimensions: 3600x1200+0+0
creating pixmap (3600x1200)
xwallpaper: output LVDS1 was not found/disconnected, ignoring
composing /home/themad/wallpaper/pic1.jpg for DP2 (area 1920x1200+0+0) (mode 3)
xcb image (1920x1200) to DP2 (1920x1200+1680+0)
put image request parameters:
maximum request length allowed for server (32 bits): 4194303
maximum length for row data: 16777188
length of rows in image: 7680
maximum height to send: 2184
put image (1920x1200+0+0) to DP2 (1920x1200+1680+0)
composing /home/themad/wallpaper/pic2.png for VGA1 (area 1680x1050+0+0) (mode 3)
xcb image (1680x1050) to VGA1 (1680x1050+0+0)
put image request parameters:
maximum request length allowed for server (32 bits): 4194303
maximum length for row data: 16777188
length of rows in image: 6720
maximum height to send: 2496
put image (1680x1050+0+0) to VGA1 (1680x1050+0+0)
xwallpaper: output HDMI1 was not found/disconnected, ignoring
xwallpaper: output DP1 was not found/disconnected, ignoring
xwallpaper: output HDMI2 was not found/disconnected, ignoring
event received: response_type=89, sequence=30
event received: response_type=89, sequence=30
event received: response_type=89, sequence=60
```

**Comment by stoeckmann** [comment]
Great, thank you for the explanation! I can use it to reproduce the issue and get back to you when it's properly solved.

**Comment by stoeckmann** [comment]
I have released xwallpaper 0.6.5 which fixes this issue.

At first I thought about being clever and also react on CRT change events to reduce the amount of redrawings. But in the end, I am better off just listening on screen change events. Before I try to outsmart RandR event handling, I simply use their information.

Let me know if everything works (I was able to reproduce the issue and also that it works on my setup) -- and ideally close the issue if it's resolved. :)
